scripts/benchmark_all.sh

```diff
@@ -12,7 +12,17 @@ then
 fi
 
 benchmarks='hyriseBenchmarkTPCH hyriseBenchmarkTPCDS hyriseBenchmarkTPCC hyriseBenchmarkJoinOrder'
-num_mt_clients=50
+warmup_seconds=2
+runs=100
+
+# Setting the number of clients used for the multi-threaded scenario to the machine's core count.
+# This only works for macOS and Linux.
+output="$(uname -s)"
+case "${output}" in
+    Linux*)     num_mt_clients="$(lscpu -p | egrep -v '^#' | grep '^[0-9]*,[0-9]*,0,0' | wc -l)";;
+    Darwin*)    num_mt_clients="$(sysctl -n hw.ncpu)";;
```

Comment:
Bash syntax ... you have to love it.

Reply:
Every time I think that they really optimize for the worst possible syntax.